src/fsharp/TypeChecker.fs

```diff
@@ -5977,7 +5979,7 @@ and TcExprUndelayed cenv overallTy env tpenv (expr: SynExpr) =
     | SynExpr.TraitCall(tps,memSpfn,arg,m) ->
         let synTypes =  tps |> List.map (fun tp -> SynType.Var(tp,m))
         let (TTrait(_,logicalCompiledName,_,argtys,returnTy,_) as traitInfo),tpenv = TcPseudoMemberSpec cenv NewTyparsOK env synTypes tpenv memSpfn m
-        if List.contains logicalCompiledName BakedInTraitConstraintNames then 
+        if BakedInTraitConstraintNames.Contains logicalCompiledName then 
```

Comment:
Practically speaking I'm pretty certain HashSet is safe when used as an immutable collection (concurrent readers with no further writers).  I also suspect we rely on that kind of property elsewhere (i.e. "mutable collection becomes threadsafe after last write") 

But it's true that the documentation doesn't say that specifically...  Maybe use an System.Collections.Immutable collection explicitly.